metricbeat/module/kubernetes/util/kubernetes.go

```diff
@@ -161,13 +161,7 @@ func NewResourceMetadataEnricher(
 		return &nilEnricher{}
 	}
 
-	// GetPodMetaGen requires cfg of type Config
-	commonMetaConfig := metadata.Config{}
-	if err := base.Module().UnpackConfig(&commonMetaConfig); err != nil {
```

Comment:
I might miss smth here, but I am not sure why it was used here. `cfg` is used later to pass it to [`metadata.GetPodMetaGen`](https://github.com/elastic/beats/pull/37458/files#diff-1d7b17f1ef239e6177db5ee660f2234bfd69c82e93494d4e46ee97f8a8617cfdR192), `metadata.NewServiceMetadataGenerator` and `metadata.NewNamespaceAwareResourceMetadataGenerator`, as result we get the default `metadata.Config`:
<img width="1601" alt="Screenshot 2023-12-20 at 15 22 48" src="https://github.com/elastic/beats/assets/28299531/87e9da3c-0bdb-4caa-bbf7-767fe041221a">

All those functions are using `cfg` in `NewResourceMetadataGenerator` - where happens the same: the https://github.com/elastic/elastic-agent-autodiscover/blob/main/kubernetes/metadata/resource.go#L46

Reply:
@MichaelKatsoulis as discussed - I've reverted this change and added missing metadata enrichment parameters to the configs https://github.com/elastic/beats/pull/37458/commits/0215971e6ec689b794a4e1c7d970a5ab7c79b01a